store/postgres/src/store.rs

```diff
@@ -316,102 +301,65 @@ impl Store {
-                    .bind::<Text, _>(order_attribute)
-                    .sql("->> 'data')")
-                    .sql(cast_type)
-                    .sql(" ")
-                    .sql(direction)
-                    .sql(" NULLS LAST"),
-            );
-        }
-
-        // Add range filter to query
-        if let Some(limit) = query.range.first {
-            diesel_query = diesel_query.limit(limit as i64);
-        }
-        if query.range.skip > 0 {
-            diesel_query = diesel_query.offset(query.range.skip as i64);
-        }
-
-        // Finally add the selected columns
-        let diesel_query = diesel_query.select((data, entity));
-
-        // Record debug info in case of error
-        let diesel_query_debug_info = debug_query(&diesel_query).to_string();
+        let order = match query.order_by {
+            Some((attribute, value_type)) => {
+                let direction = query
+                    .order_direction
+                    .map(|direction| match direction {
+                        EntityOrder::Ascending => "ASC",
+                        EntityOrder::Descending => "DESC",
+                    })
+                    .unwrap_or("ASC");
+                let cast_type = match value_type {
+                    ValueType::BigInt | ValueType::BigDecimal => "::numeric",
+                    ValueType::Boolean => "::boolean",
+                    ValueType::Bytes => "",
+                    ValueType::ID => "",
+                    ValueType::Int => "::bigint",
+                    ValueType::String => "",
+                    ValueType::List => {
+                        return Err(QueryExecutionError::OrderByNotSupportedForType(
+                            "List".to_string(),
+                        ));
+                    }
+                };
+                Some((attribute, cast_type, direction))
+            }
+            None => None,
+        };
 
         // Process results; deserialize JSON data
-        diesel_query
-            .load::<(serde_json::Value, String)>(conn)
-            .map(|values| {
-                values
-                    .into_iter()
-                    .map(|(value, entity_type)| {
-                        let parse_error_msg = format!("Error parsing entity JSON: {:?}", data);
-                        let mut value =
-                            serde_json::from_value::<Entity>(value).expect(&parse_error_msg);
-                        value.set("__typename", entity_type);
-                        value
-                    })
-                    .collect()
-            })
-            .map_err(|e| {
-                QueryExecutionError::ResolveEntitiesError(format!(
-                    "{}, query = {:?}",
-                    e, diesel_query_debug_info
-                ))
-            })
+        conn.query(
+            &query.subgraph_id,
+            query.entity_types,
+            query.filter,
+            order,
+            query.range.first.unwrap_or(0),
```

Comment:
The default for first is 100, not 0.

Reply:
Should it be mandatory here? Isn't this that case where it would cause our internal unlimited queries to have a limit of 100?

Reply:
We have internal queries that can have arbitrary complexity but at least in the dynamic data sources loading code, we paginate based on the assumption that the limit is `100` per collection.

Reply:
I was a little overzealous in making the `entities` API low-level and barebones. `0` signals 'do not restrict the number of entities'. I'll just pass the original `Option` through to make that more obvious.